src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditorViewModel.java

```diff
@@ -0,0 +1,286 @@
+import org.jabref.logic.net.URLDownload;
+import org.jabref.logic.util.OS;
+import org.jabref.logic.util.io.FileFinder;
+import org.jabref.logic.util.io.FileFinders;
+import org.jabref.logic.util.io.FileUtil;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FileFieldParser;
+import org.jabref.model.entry.FileFieldWriter;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.preferences.JabRefPreferences;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class LinkedFilesEditorViewModel extends AbstractEditorViewModel {
+    private static final Log LOGGER = LogFactory.getLog(LinkedFilesEditorViewModel.class);
+
+    private ListProperty<LinkedFileViewModel> files = new SimpleListProperty<>(FXCollections.observableArrayList(LinkedFileViewModel::getObservables));
+    private BooleanProperty fulltextLookupInProgress = new SimpleBooleanProperty(false);
+    private DialogService dialogService;
+    private BibDatabaseContext databaseContext;
+    private TaskExecutor taskExecutor;
+    public LinkedFilesEditorViewModel(DialogService dialogService, BibDatabaseContext databaseContext, TaskExecutor taskExecutor) {
+        this.dialogService = dialogService;
+        this.databaseContext = databaseContext;
+        this.taskExecutor = taskExecutor;
+        BindingsHelper.bindContentBidirectional(
+                files,
+                text,
+                LinkedFilesEditorViewModel::getStringRepresentation,
+                this::parseToFileViewModel
+        );
+    }
+
+    private static String getStringRepresentation(List<LinkedFileViewModel> files) {
+        // Only serialize linked files, not the ones that are automatically found
+        List<LinkedFile> filesToSerialize = files.stream()
+                .filter(file -> !file.isAutomaticallyFound())
+                .map(LinkedFileViewModel::getFile)
+                .collect(Collectors.toList());
+
+        return FileFieldWriter.getStringRepresentation(filesToSerialize);
+    }
+
+    public boolean isFulltextLookupInProgress() {
+        return fulltextLookupInProgress.get();
+    }
+
+    public BooleanProperty fulltextLookupInProgressProperty() {
+        return fulltextLookupInProgress;
+    }
+
+    private List<LinkedFileViewModel> parseToFileViewModel(String stringValue) {
+        return FileFieldParser.parse(stringValue).stream()
+                .map(LinkedFileViewModel::new)
+                .collect(Collectors.toList());
+    }
+
+    public ObservableList<LinkedFileViewModel> getFiles() {
+        return files.get();
+    }
+
+    public ListProperty<LinkedFileViewModel> filesProperty() {
+        return files;
+    }
+
+    public void addNewFile() {
+        List<Path> fileDirectories = databaseContext.getFileDirectoriesAsPaths(Globals.prefs.getFileDirectoryPreferences());
+        Path workingDirectory = fileDirectories.stream()
```

Comment:
There is a method I once crated: getFirstFileDirectory (or similar) which exactly does this. I think it is in databaseContext, too